eos/scattering/ee-to-ccbar.cc

```diff
@@ -0,0 +1,778 @@
+                    "e^+e^-->ccbar"
+                    )
+                );
+        }
+
+        const IntermediateResult * prepare(const complex<double> & E)
+        {
+            // Amplitude on the first RS
+            _intermediate_result.tmatrix_row_0 = K->tmatrix_row(0, E * E);
+            // Amplitude on the second RS
+            _intermediate_result.tmatrix2_row_0 = K->tmatrix_row(0, E * E, true);
+
+            _intermediate_result.E = E;
+            _intermediate_result.s = E * E;
+
+            return &_intermediate_result;
+        }
+
+
+        double rho(const IntermediateResult * intermediate_result, const Channels & channel)
+        {
+            return std::real(K->_channels[Channels(channel)]->rho(intermediate_result->s));
+        }
+
+        complex<double> chew_mandelstam(const IntermediateResult * intermediate_result, const Channels & channel)
+        {
+            return K->_channels[Channels(channel)]->chew_mandelstam(intermediate_result->s);
+        }
+
+        complex<double> chew_mandelstam_II(const IntermediateResult * intermediate_result, const Channels & channel)
+        {
+            complex<double> s = intermediate_result->s;
+            auto get_channel = K->_channels[Channels(channel)];
+
+            const unsigned li = get_channel->_l_orbital;
+            const double q0 = get_channel->_q0.evaluate();
+            const complex<double> mi1_2 = power_of<2>((double)get_channel->_m1);
+            const complex<double> mi2_2 = power_of<2>((double)get_channel->_m2);
+
+            // Momentum of particles in their center-of-momentum frame
+            const complex<double> q = 0.5 * sqrt(eos::lambda(s, mi1_2, mi2_2)) / sqrt(s);
+
+            // Blatt-Weisskopf factors, cf eq. (50.26)
+            const complex<double> Fi = kmatrix_utils::blatt_weisskopf_factor(li, q / q0);
+
+            return get_channel->chew_mandelstam(s) +
+                complex<double>(0.0, 2.0) * get_channel->rho(s) * power_of<2>(pow(q / q0, li) * Fi);
+        }
+
+
+        inline double sigma_eetomumu(const double & E)
+        {
+            // Conversion factor between GeV^2 and nb
+            const double speedoflight = 299792458.; // Exact value
+            const double GeVtonb = 10 * power_of<2>( 1.e18 * hbar * speedoflight);
+
+            return GeVtonb * 4.0 * M_PI * alpha_em * alpha_em / (3.0 * E*E);
+        }
+
+        // amplitude of ee -> channel
+        complex<double> amplitude_eetochannel(const IntermediateResult * intermediate_result, const Channels & channel)
+        {
+            // Conversion factor between GeV^2 and nb
+            const double speedoflight = 299792458.; //Exact value
+            const double GeVtonb = 10 * power_of<2>( 1.e18 * hbar * speedoflight);
+
+            // Channel properties
+            const double Nf = 2 * K->_channels[Channels(channel)]->_l_orbital + 1;
```

Comment:
All number literals in an expression should be double literals, i.e.m ``2.0`` and ``1.0``.